### ISSUE TITLE
Custom grid partial - Uncaught TypeError: Cannot read property 'element' of undefined

### ISSUE THREAD

**Issue by mkchmiel** [NonFactors/MVC5.Grid#140]
Hi,

I seem to be getting the Uncaught TypeError: Cannot read property 'element' of undefined message within chrome when using the custom grid partial. The data appears to load without any issues, as of right now this appears more of a nuisance than a functionality breaking issue. What information do you need to help diagnose the problem?

**Comment by Muchiachio** [comment]
Your grid partial probably includes non-grid elements or the root of the dom is not a mvc-grid div.

**Comment by mkchmiel** [comment]
Main CSHTML
<div class="row" style="padding-top: 1em;">
    <div class="col-5">
        <h2>Sales Collection - @ViewBag.WarehouseDesc</h2>
    </div>
    <div class="col-5">
        <h2>@ViewBag.ReasonCode - @ViewBag.ReasonCodeDesc</h2>
    </div>
    <div class="col-2">
        <button class="float-right mega-octicon octicon-fold" style="margin-left:10px;" onclick="hideAll();"></button>
        <button class="float-right mega-octicon octicon-unfold" style="margin-left:10px;" onclick="showAll();"></button>
    </div>
    
</div>



@Html.AjaxGrid(Url.Action("SalesCollectionGrid", new { businessUnit = ViewBag.BusinessUnit, warehouse = ViewBag.Warehouse, selectedDate = ViewBag.SelectedDate, reasonCode = ViewBag.ReasonCode, reasonCodeDesc = ViewBag.ReasonCodeDesc }))

@section Scripts
{
    <script>
        $('.mvc-grid').mvcgrid({
            //The below allows us to show a loading icon
            /*reloadStarted: function (grid) {
                document.getElementById("loader").style.display = "block";
            },
            reloadEnded: function (grid) {
                document.getElementById("loader").style.display = "none";
            }*/
        });

        function showAll() {
            $('.collapse').collapse('show');
            var test = $('.accordionDropDown');
            test.removeClass('accordionDropDown');
            test.addClass('accordionDropDown2');
        }
        function hideAll() {
            $('.collapse').collapse('hide');
            var test = $('.accordionDropDown2');
            test.removeClass('accordionDropDown2');
            test.addClass('accordionDropDown');
        }

        //There must be a better way to do this, till than it will work
        function toggleIcon(e) {
            var targetById = document.getElementById(e);
            if (targetById.classList.contains('accordionDropDown')) {
                targetById.classList.remove('accordionDropDown');
                targetById.classList.add('accordionDropDown2');
            }
            else {
                targetById.classList.remove('accordionDropDown2');
                targetById.classList.add('accordionDropDown');
            }
        }
    </script>
}

E:\Solutions\Web\RBGCustomPortal\RBGCustomPortal\Views\Adjustments\Partials\_SalesCollectionGrid.cshtml
@model IEnumerable<RBGCustomPortal.Models.Supervisor>

@*@Html.ValidationSummary(true, "", new { @class = "text-danger" })*@
@(Html
    .Grid("Partials/_SalesCollectionNestedGrid", Model)
    .Build(columns =>
    {
        columns.Add(model => model.SupervisorDesc).Titled("Supervisor");
        columns.Add(model => model.Accounts).Titled("Adjusted Accounts");
        columns.Add(model => model.CE).Titled("Adjusted CE");
    })
    .Filterable()
    .Sortable()
    .Empty("Sorry, no data found.")
    .WithFooter("Partials/_SalesGridFooter")
    .Css("table table-sm table-striped table-hover")
)

E:\Solutions\Web\RBGCustomPortal\RBGCustomPortal\Views\Adjustments\Partials\_SalesCollectionNestedGrid.cshtml
@using RBGCustomPortal.Models
@model IGrid<Supervisor>

<div class="mvc-grid">
    <table>
        <thead>
            <tr class="headerRow ">
                @foreach (IGridColumn column in Model.Columns)
                {
                    <th>
                        <span class="mvc-grid-header-title">@column.Title</span>
                    </th>
                }
            </tr>
        </thead>

        @if (Model.Rows.Any())
        {
            <tbody>
                @foreach (IGridRow<Supervisor> row in Model.Rows)
                {
                    var rowID = "#salesRepList" + row.Model.SupervisorCode.TrimEnd();
                    <tr class="@row.CssClasses" data-toggle="collapse" role="button" aria-controls="@rowID.TrimStart('#')" aria-expanded="false" data-target="@rowID" onclick="toggleIcon('supervisor @row.Model.SupervisorCode.Trim()')">
                        @{ var lastSupervisorColumn = Model.Columns.Last();}
                        @foreach (IGridColumn column in Model.Columns)
                        {
                            <td class="@(column == lastSupervisorColumn ? "accordionDropDown" : "")" id="supervisor @(column == lastSupervisorColumn ? row.Model.SupervisorCode.Trim() : "")">@column.ValueFor(row)</td>
                        }
                    </tr>

                    <tr class="collapse" id="@rowID.TrimStart('#')">
                        <td colspan="3">
                            <div class="mvc-grid" data-filter-mode="ExcelRow" data-source-url="" style="padding-left:10px; padding-right:10px;">
                                <table>
                                    <thead class="mvc-grid-headers">
                                        <tr>
                                            <th>
                                                <span class="mvc-grid-header-title ">Sales Rep</span>
                                            </th>
                                            <th>
                                                <span class="mvc-grid-header-title ">Accounts</span>
                                            </th>
                                            <th>
                                                <span class="mvc-grid-header-title ">CE</span>
                                            </th>
                                        </tr>
                                    </thead>
                                    <tbody>
                                        @foreach (var item in row.Model.SalesRepList)
                                        {
                                            var @salesRepID = "#accountList" + item.SalesRepCode;
                                            <tr class="@row.CssClasses" data-toggle="collapse" role="button" data-patent="@salesRepID" aria-controls="@salesRepID.TrimStart('#')" data-target="@salesRepID" onclick="toggleIcon('sales @item.SalesRepCode.Trim()')">
                                                <td>@item.SalesRepDesc</td>
                                                <td>@item.Accounts</td>
                                                <td class="accordionDropDown" id="sales @item.SalesRepCode.Trim()">@item.CE</td>
                                            </tr>

                                            <tr class="collapse" id="@salesRepID.TrimStart('#')">
                                                <td colspan="7">
                                                    <div class="mvc-grid" data-name="" data-filter-mode="ExcelRow" data-source-url="" style="padding-left:10px; padding-right:10px;">
                                                        <table>
                                                            <thead>
                                                                <tr class="mvc-grid-headers">
                                                                    <th>
                                                                        <span class="mvc-grid-header-title ">Load</span>
                                                                    </th>
                                                                    <th>
                                                                        <span class="mvc-grid-header-title ">Driver Code</span>
                                                                    </th>
                                                                    <th>
                                                                        <span class="mvc-grid-header-title ">Driver Name</span>
                                                                    </th>
                                                                    <th>
                                                                        <span class="mvc-grid-header-title ">Account</span>
                                                                    </th>
                                                                    <th>
                                                                        <span class="mvc-grid-header-title ">Business Name</span>
                                                                    </th>
                                                                    <th>
                                                                        <span class="mvc-grid-header-title ">Address</span>
                                                                    </th>
                                                                    <th>
                                                                        <span class="mvc-grid-header-title ">City</span>
                                                                    </th>
                                                                </tr>
                                                            </thead>
                                                            <tbody>
                                                                @foreach (var item2 in item.AccountList)
                                                                {
                                                                    var @accountCode = "#account" + item2.AccountCode;
                                                                    <tr class="@row.CssClasses" data-toggle="collapse" role="button" data-patent="@accountCode" aria-controls="@accountCode.TrimStart('#')" data-target="@accountCode" onclick="toggleIcon('account @item2.AccountCode')">
                                                                        <td>@item2.Load</td>
                                                                        <td>@item2.Driver</td>
                                                                        <td>@item2.DriverDesc</td>
                                                                        <td>@item2.AccountCode</td>
                                                                        <td>@item2.BusinessName</td>
                                                                        <td>@item2.Address</td>
                                                                        <td class="accordionDropDown" id="account @item2.AccountCode">@item2.City</td>
                                                                    </tr>

                                                                    <tr class="collapse" id="@accountCode.TrimStart('#')">
                                                                        <td colspan="8">
                                                                            <div class="mvc-grid" data-name="" data-filter-mode="ExcelRow" data-source-url="" style="padding-left:10px; padding-right:10px;">
                                                                                <table>
                                                                                    <thead>
                                                                                        <tr class="mvc-grid-headers">
                                                                                            <th>
                                                                                                <span class="mvc-grid-header-title ">Operator ID</span>
                                                                                            </th>
                                                                                            <th>
                                                                                                <span class="mvc-grid-header-title ">Location ID</span>
                                                                                            </th>
                                                                                            <th>
                                                                                                <span class="mvc-grid-header-title ">Item Code</span>
                                                                                            </th>
                                                                                            <th>
                                                                                                <span class="mvc-grid-header-title ">Description</span>
                                                                                            </th>
                                                                                            <th>
                                                                                                <span class="mvc-grid-header-title ">Quantity Requested</span>
                                                                                            </th>
                                                                                            <th>
                                                                                                <span class="mvc-grid-header-title ">Quantity Picked</span>
                                                                                            </th>
                                                                                            <th>
                                                                                                <span class="mvc-grid-header-title ">Zone ID</span>
                                                                                            </th>
                                                                                            <th>
                                                                                                <span class="mvc-grid-header-title ">Pick Time</span>
                                                                                            </th>
                                                                                        </tr>
                                                                                    </thead>
                                                                                    <tbody>
                                                                                        @foreach (var item3 in item2.ProductList)
                                                                                        {
                                                                                            <tr>
                                                                                                <td>@item3.OperatorID</td>
                                                                                                <td>@item3.LocationID</td>
                                                                                                <td>@item3.ItemCode</td>
                                                                                                <td>@item3.Description</td>
                                                                                                <td>@item3.QuantityRequested</td>
                                                                                                <td>@item3.QuantityPicked</td>
                                                                                                <td>@item3.ZoneID</td>
                                                                                                <td>@item3.PickTime</td>
                                                                                            </tr>
                                                                                        }
                                                                                    </tbody>
                                                                                    <tfoot>

                                                                                    </tfoot>
                                                                                </table>
                                                                            </div>

                                                                        </td>
                                                                    </tr>
                                                                }
                                                            </tbody>
                                                        </table>
                                                    </div>
                                                </td>
                                            </tr>
                                        }
                                    </tbody>
                                    <tfoot>
                                        <tr>
                                            <td><b>Totals</b></td>
                                            <td><b>@((row.Model.SalesRepList.Sum(row2 => row2.Accounts)))</b></td>
                                            <td><b>@((row.Model.SalesRepList.Sum(row2 => row2.CE)))</b></td>
                                        </tr>
                                    </tfoot>
                                </table>
                            </div>
                        </td>
                    </tr>
                }
            </tbody>
            <tfoot>
                <tr>
                    <td><b>Totals</b></td>
                    <td><b>@(Model.Rows.Sum(row => row.Model.Accounts))</b></td>
                    <td><b>@((Model.Rows.Sum(row => row.Model.CE)))</b></td>
                </tr>
            </tfoot>
        }
        else
        {
            <tbody>
                <tr class="mvc-grid-empty-row">
                    <td colspan="@Model.Columns.Count()">
                        @Model.EmptyText
                    </td>
                </tr>
            </tbody>
        }
    </table>
</div>

**Comment by Muchiachio** [comment]
Sorry, but it takes too much time to make a repro from a code mash.
You can check the failing line in `mvc-grid.js`. It has a rendered html nearby, so you can check the html.

**Comment by mkchmiel** [comment]
Will do, thank you.

**Comment by mkchmiel** [comment]
Just in case anyone ever comes back here with the same issue, I resolved this by removing a comment I had in the E:\Solutions\Web\RBGCustomPortal\RBGCustomPortal\Views\Adjustments\Partials_SalesCollectionNestedGrid.cshtml.

Ironically enough I had removed it when pasting it here as it was a personal note. Long story short the answer is exactly what Muchiachio pointed out. "Your grid partial probably includes non-grid elements or the root of the dom is not a mvc-grid div.". A comment can break this.
